int32 goal
float64 time
---
int32 newgoal